test/services/users.test.ts

```diff
@@ -1,13 +1,237 @@
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code, message } = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Email is already in use. Please log in');
+      }
+    });
+
+    it('should fail - missed data', async () => {
+
+      try{
+        await service.create({
+          lastName: 'test user last',
+          email: 'create_user@example.com',
+        });
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code, message } = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('Invalid data');
+      }
+    });
+
+  });
+
+  describe('get user', () => {
+    before(async () => {
+      const { id } = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - get user by id', async () => {
+      const { email, id } = await service.get(userId);
+      expect(id).to.exist;
+      expect(email).to.equal(userInfo.email);
+    });
+
+    it('should fail - not existed userId', async () => {
+      try{
+        await service.get(unexistedId);
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code } = err;
+        expect(code).to.equal(404);
+      }
+    });
+
+    it('should fail - invalid userId', async () => {
+      try{
+        await service.get('123456');
+        expect.fail('call should have failed');
+      } catch (err: any) {
+        const { code, message } = err;
+        expect(code).to.equal(400);
+        expect(message).to.equal('ValidationError: "id" must be a valid GUID');
+      }
+    });
+  });
+
+  describe('find user', () => {
+    before(async () => {
+      const { id } = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - find user', async () => {
+      const response = await service.find();
+      const { total } = response;
+      expect(total).not.to.equal(0);
+      expect(response).to.have.keys(['total', 'limit', 'skip', 'data']);
+
+    });
+  });
+
+  describe('update (patch) user', () => {
+    const updateUser ={
+      firstName: 'updated first name',
+      lastName: 'updated user last name',
+    };
+    const updateUserEmail ={
+      email: 'updatedemail@test.com'
+    };
+    before(async () => {
+      const { id } = await service.create(userInfo);
+      userId = id;
+    });
+    after(async () => {
+      await service.remove(userId);
+    });
+
+    it('should succeed - update(patch) user', async () => {
+      await service.patch(userId, updateUser);
+      const { firstName, lastName } = await service.get(userId);
+      expect(firstName).to.equal(updateUser.firstName);
+      expect(lastName).to.equal(updateUser.lastName);
+    });
+
+    it('should succeed - update(patch) user email', async () => {
```

Comment:
Add one more test - `should fail - email already taken`

Reply:
test added e63ee2f79bcab0301e1774bea05285cfad975b6d